packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -489,11 +489,20 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
       assert(minScrollExtent != null);
       assert(maxScrollExtent != null);
       assert(minScrollExtent <= maxScrollExtent);
+      final bool isFirstApplication = _minScrollExtent == null || _maxScrollExtent == null;
       _minScrollExtent = minScrollExtent;
       _maxScrollExtent = maxScrollExtent;
       _haveDimensions = true;
-      applyNewDimensions();
```

Comment:
@gaaclarke was there a particular thing that led to you inlining applyNewDimensions here, as opposed to letting people override it?

Reply:
er, i mean `approveNewDimensions`

Reply:
ah, i see -- that was a response to my comment above.